examples/SampleCustomLoggerCApi.c

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <pulsar/c/client.h>
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+void format_time(char *output){
+    time_t rawtime;
+    struct tm * timeinfo;
+
+    time(&rawtime);
+    timeinfo = localtime(&rawtime);
+
+    sprintf(output, "%d %d %d %d:%d:%d",
+            timeinfo->tm_year + 1900, timeinfo->tm_mon + 1,  timeinfo->tm_mday,
+            timeinfo->tm_hour, timeinfo->tm_min, timeinfo->tm_sec);
+}
+
+void custom_logger(pulsar_logger_level_t level, const char *file, int line, const char *message,
+              void *ctx) {
+    time_t mytime = time(NULL);
+    char * time_str = ctime(&mytime);
+    // Control the log level yourself.
+    if (level >= pulsar_DEBUG) {
+        format_time(time_str);
+        printf("[%s] [%u] [%s] [%d] [%s] \n", time_str, level, file, line, message);
+    }
```

Comment:
Though it's an example, we should check `level` first to avoid `time` and `ctime` calls.